master/Master.hs

```diff
@@ -91,42 +99,38 @@ timeoutThread = do
   liftIO $ threadDelay 10000000
   timeoutThread
 
+addVote :: KVTxnId -> Int -> ReaderT (MVar MasterState) IO ()
+addVote tid sid = ask >>= \mvar -> liftIO $ do
+  s <- takeMVar mvar
+  let v = fromJust $ Map.lookup tid (voteMap s)
+  putMVar mvar s { voteMap = Map.insert tid (Set.insert sid v) $ voteMap s }
+  traceIO (show s)
+
+votingComplete :: KVTxnId -> ReaderT (MVar MasterState) IO Bool
+votingComplete tid = ask >>= \mvar -> liftIO $ do
+  s <- readMVar mvar
+  let v = fromJust $ Map.lookup tid (voteMap s)
+  return $ Set.size v == Prelude.length (Lib.slaveConfig $ cfg s)
+
+timedOut :: KVTxnId -> ReaderT (MVar MasterState) IO Bool
+timedOut tid = ask >>= \mvar -> liftIO $ do
+  s <- readMVar mvar
+  now <- Utils.currentTimeInt
+  let (ts, _) = fromJust $ Map.lookup tid (timeoutMap s) -- TODO: guarantee that a timeout value will exist  
```

Comment:
I believe it should exist, but when I execute this code, I get a `fromJust` exception in the `votingComplete` phase.. I think this could happen if two `ACKS` are sent by slave for the same commit (which is possible)